File: 0376 - Wiggle Subsequence/solution.py
class Solution:
    def wiggleMaxLength(self, A: List[int]) -> int:
        up = down = 1
        for i in range(len(A)-1):
            if A[i+1] > A[i]:
                up = down+1
            elif A[i+1] < A[i]:
                down = up+1
        return max(up, down)

File: 0376 - Wiggle Subsequence/solution.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& A) {
        int n = A.size(), up = 1, down = 1;
        for(int i = 1; i < n; i++){
            if(A[i] > A[i-1]) up = down+1;
            else if(A[i] < A[i-1]) down = up+1;
        }
        return max(up, down);
    }
};